### PR TITLE
Create a config tree per configuration layers

### PR DESCRIPTION
**Pull request by hush-hush**
### What does this PR do?

Create a config tree per configuration layers. This PR also update the `Set` method in another commit to handle all layers.

### CODE REVIEW

File: pkg/config/nodetreemodel/config.go [hunk 1]
```diff
@@ -139,18 +154,35 @@ func (c *ntmConfig) Set(key string, newValue interface{}, source model.Source) {
...
+		tree = c.cli
 	}
 
 	c.Lock()
 
 	previousValue, _ := c.getValue(key)
-	_, _ = c.set(key, newValue, tree, source)
-	updated, _ := c.set(key, newValue, c.root, source)
+
+	parts := strings.Split(strings.ToLower(key), ",")
```
Comment (dustmop) [inline pkg/config/nodetreemodel/config.go:179]: You can use `splitKey` here once https://github.com/DataDog/datadog-agent/pull/30835 merges.

File: pkg/config/nodetreemodel/config.go [hunk 2]
```diff
@@ -139,18 +154,35 @@ func (c *ntmConfig) Set(key string, newValue interface{}, source model.Source) {
...
+	_, _ = tree.SetAt(parts, newValue, source)
+	updated, _ := c.root.SetAt(parts, newValue, source)
 
 	// if no value has changed we don't notify
 	if !updated || reflect.DeepEqual(previousValue, newValue) {
+		c.Unlock()
```
Comment (dustmop) [inline pkg/config/nodetreemodel/config.go:185]: I fixed this in the other PR by moving the Unlock before this conditional.

**Review by dustmop — APPROVED** [review]
LGTM!

File: pkg/config/nodetreemodel/config.go [hunk 3]
```diff
@@ -139,18 +154,35 @@ func (c *ntmConfig) Set(key string, newValue interface{}, source model.Source) {
...
 	}
 
 	c.Lock()
 
 	previousValue, _ := c.getValue(key)
-	_, _ = c.set(key, newValue, tree, source)
-	updated, _ := c.set(key, newValue, c.root, source)
+
+	parts := strings.Split(strings.ToLower(key), ",")
+	_, _ = tree.SetAt(parts, newValue, source)
```
Comment (GustavoCaso) [inline pkg/config/nodetreemodel/config.go:180]: Should we check the return error to ensure nothing went wrong?

File: pkg/config/nodetreemodel/config.go [hunk 4]
```diff
@@ -139,18 +154,35 @@ func (c *ntmConfig) Set(key string, newValue interface{}, source model.Source) {
...
+	updated, _ := c.root.SetAt(parts, newValue, source)
```
Comment (GustavoCaso) [inline pkg/config/nodetreemodel/config.go:181]: Same as above 😄

File: pkg/config/nodetreemodel/config_test.go [hunk 5]
```diff
@@ -25,3 +28,121 @@ func TestBuildDefaultMakesTooManyNodes(t *testing.T) {
...
+	assert.Equal(t, 3, cfg.Get("env"))
+	assert.Equal(t, 4, cfg.Get("runtime"))
+	assert.Equal(t, 5, cfg.Get("localConfigProcess"))
+	assert.Equal(t, 6, cfg.Get("rc"))
+	assert.Equal(t, 7, cfg.Get("fleetPolicies"))
+	assert.Equal(t, 8, cfg.Get("cli"))
+
+	assert.Equal(t, model.SourceDefault, cfg.GetSource("default"))
+	assert.Equal(t, model.SourceUnknown, cfg.GetSource("unknown"))
+	assert.Equal(t, model.SourceFile, cfg.GetSource("file"))
```
Comment (GustavoCaso) [inline pkg/config/nodetreemodel/config_test.go:121]: We do not seems to set the `"file"` config as the rest in lines 101-107. Is that expected?

Comment (hush-hush) [reply pkg/config/nodetreemodel/config_test.go:200]: It's set by `ReadConfig` method who read a yaml payload.

**Review by GustavoCaso — APPROVED** [review]

**Pull request merged**